src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -899,7 +900,22 @@ private void CreateGroup(object parameters)
 
         private bool CanCreateGroup(object parameters)
         {
-            return DynamoSelection.Instance.Selection.OfType<ModelBase>().Any();
+            var groups = WorkspaceViewModel.Model.Annotations;
+            //Create Group should be disabled when a group is selected
+            if (groups != null && groups.Any(x => x.IsSelected))
+            {
+                return false;
+            }
+
+            //Create Group should be disabled when a node selected is already in a group
+            if (groups != null && !groups.Any(x => x.IsSelected))
+            {               
+                return !((from model in groups
+                    where model.SelectedModels.Any(x => x.GUID == this.NodeLogic.GUID)
+                    select model).Any());
```

Comment:
The mixing of different kinds of LINQ syntax feels kinda weird.  How about this?:

```
groups.SelectRange(m => m.SelectedModels ).Any(m => m.GUID == NodeLogic.GUID );
```